### PR TITLE
Stop using catullus.pl to generate the DATA lines in pg_type.h

### PR DESCRIPTION
**Pull request by hlinnaka**
We only have a few extra types in GPDB, so it doesn't seem worth maintaining
the catullus mechanism for updating pg_type.h. When a new column is added
in the upstream, we'll just fix the GPDB-added lines manually. There are no
GPDB-added columns in pg_type.h, so we don't need the GPDB_COLUMN_DEFAULT
tricks we used in pg_proc.h.

Catullus.pl still needs type information to map the type names in
pg_proc_gp.sql to type OIDs. It used to get that information from the
pg_type.sql file, but I changed it to read the DATA lines in pg_type.h
directly.

Also revert unnecessary whitespace and comment changes against upstream, to
make merge easier.

### CODE REVIEW

File: src/include/catalog/catullus.pl [hunk 1]
```diff
@@ -1509,673 +1462,33 @@ ()
...
-	my %h1 = @_;
-
-	die ("no oid")
-		unless (exists($h1{with}) &&
-				exists($h1{with}->{oid}));
+	my $fh;
 
-	die ("no tuple")
-		unless (exists($h1{tuple}));
+	open $fh, "< pg_type.h"
```
Comment (danielgustafsson) [inline src/include/catalog/catullus.pl]: While I know this is style copied from the removed hunk it's an old and non-recommended way of writing Perl. The modern way would be the three-argument form:
```Perl
open(my $fh, "<", "pg_type.h")
```

**Comment by danielgustafsson** [comment]
Yes please, :+1:! The small style nitpick can be left if you want, I intend to go over all the Perl code at some point and modernize them.

**Comment by hlinnaka** [comment]
Pushed. I left the style nitpick in place for you ;-)

**Comment by kuien** [comment]
Cool! And we might need to update the README.xxx under catalog dir as well.

**Comment by hlinnaka** [comment]
On 11/03/16 17:03, Kuien Liu wrote:
> Cool! And we might need to update the README.xxx under catalog dir as well.

Ah yes, good point. Updated, thanks!

- Heikki

**Pull request closed without merging**